ESP8266Interface.cpp

```diff
@@ -46,8 +46,16 @@ int ESP8266Interface::connect(const char *ssid, const char *pass, nsapi_security
 
 int ESP8266Interface::connect()
 {
+    _esp.setTimeout(ESP8266_MISC_TIMEOUT);
+    
+    if(!_esp.check_firmware_version()) {
+        debug("ERROR: Firmware incompatible with this driver.\
```

Comment:
Also, I would omit the specific version, maybe just say "most recent version". Otherwise maintaining that number may become annoying.